05_Datathon/diabetes_prediction_example.Rmd

```diff
@@ -0,0 +1,336 @@
+
+# Create data partition using the target variable
+train_index <- createDataPartition(reduced_data$target_variable, times = 1, p = 0.8, list = FALSE)
+
+# Split data into train and test sets, select columns that will be used in model
+train_set <- reduced_data[train_index, ] %>% select(., age, bmi, gender, ethnicity,glucose_apache,icu_type, target_variable)
+head(train_set)
+test_set <- reduced_data[-train_index, ] %>% select(., age, bmi, gender, ethnicity,glucose_apache ,icu_type, target_variable)
+head(test_set)
+```
+
+## 4. Calculate summary statistics ("Table 1")
+
+- Most studies include summary statistics as Table 1.
+- `library(tableone)` makes it easy to create these summary statistics
+- Developed by Dr Kazuki Yoshida, while a Ph.D. student at Harvard University.
+- [TableOne documentation](https://cran.r-project.org/web/packages/tableone/tableone.pdf)
+
+```{r tableone, echo=TRUE}
+library(tableone)
+
+allvars = c("bmi", "age", "gender", "ethnicity","glucose_apache","icu_type")
+catvars = c("ethnicity","gender", "icu_type")
+
+table_gossis <- CreateTableOne(vars = allvars, data = train_set, 
+                factorVars = catvars,
+                strata="target_variable")
+
+kableone(table_gossis)
+```
+
+## 5. Model Construction
+
+### 5.1 Train a random forest classifier
+
+- We will use the `randomForest` library to build a random forest model.
+- See the [randomForest documentation on CRAN](https://cran.r-project.org/web/packages/randomForest/randomForest.pdf).
+
+```{r train_model, echo=TRUE}
+# Define forest tuning parameters
+
+# 5-fold cross validation
+control <- trainControl(method='repeatedcv', 
+                        number=5, 
+                        repeats=3)
+
+# Number of variables tried at each split
+mtry <- sqrt(ncol(train_set))
+
+# Grid search is a linear search through a vector of mtry values
+tunegrid <- expand.grid(.mtry=mtry)
+
+# Create classification forest using age, bmi, and gender
+forest <- randomForest(target_variable ~ age + bmi + gender + glucose_apache, 
+              data = train_set,
+              metric = "Accuracy",
+              tuneGrid = tunegrid,
+              trControl = control)
+```
+
+### 5.2 View the variable importance
+
+- Many approaches for computing "importance".
+- `rpart` calculates a metric using the [sum of contributions to splits]((https://cran.r-project.org/web/packages/rpart/vignettes/longintro.pdf)).
+- Shapley Additive exPlanations (ShAP) values are a popular alternative.
+- Compute the marginal contribution of a feature across a combination of features.
+
+```{r importance, echo=TRUE}
+library(knitr)
+kable(forest$importance)
+```
+We can see that `glucose_apache` and `bmi` had the most predictive power.
+
+
+### 5.3 Predict diabetes mellitus in five unseen patients from test set
+
+#### Select unseen patients from testing set
+
+- Let's look five unseen patients in our test set.
+- What outcome do you expect for these patients?
```

Comment:
most clinicians won't view DM as an 'outcome'.  Consider, what DM status do you expect...